vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/operationProcessor/OperationProcessor.java

```diff
@@ -103,6 +106,15 @@ public int getIncompleteResultQueueSize() {
         }
     }
 
+    /** Returns the id of the oldest operation to be sent. */
+    public Optional<String> oldestIncompleteResultId() {
+        synchronized (monitor) {
+            return Optional.of(docSendInfoByOperationId.keySet().iterator())
```

Comment:
Why can't you just say return 
docSendInfoByOperationId.isEmpty() ? Optional.empty() : Optional.of(docSendInfoByOperationId.keySet().iterator().next)
like every normal person?  :-)

Reply:
Hehe. You don't have a bad point.